spark/src/main/scala/io/ddf/spark/content/ViewHandler.scala

```diff
@@ -51,10 +56,50 @@ class ViewHandler(mDDF: DDF) extends io.ddf.content.ViewHandler(mDDF) with IHand
     this.get(columns, ViewFormat.withName(format))
   }
 
-  override  def getRandomSampleByNum(numSamples: Int, withReplacement: Boolean,
+  override  def getRandomSampleByNum(numSamples: Long, withReplacement: Boolean,
     seed:
   Int): DDF = {
-    null.asInstanceOf[DDF]
+
+    if (numSamples < 0) {
+      throw new IllegalArgumentException("Number of samples must be larger than or equal to 0")
+    }
+
+    if (!withReplacement){
+      mDDF.getSqlHandler.sql2ddf(s"select * from ${mDDF.getSchema.getTableName} order by rand($seed) limit $numSamples")
+    } else {
+      // First, create a Spark DF that contain random numbers from 0 to num_rows - 1
+      val numRows = mDDF.getNumRows
```

Comment:
Wow, great idea. It avoids join and the param numPartitions in generating randomRDD and thus has high probability to be more efficient than my approach. I think that would just erase my work, lol.